src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -497,36 +520,88 @@ public PointSetTimes eval (
             if (edges[p] == -1) {
                 return true; // Continue to next iteration.
             }
-            edge.seek(edges[p]);
-            int t1 = Integer.MAX_VALUE;
-            int t2 = Integer.MAX_VALUE;
-            // TODO this is not strictly correct when there are turn restrictions onto the edge this is linked to
-            if (distanceTableToVertices.containsKey(edge.getFromVertex())) {
-                t1 = distanceTableToVertices.get(edge.getFromVertex()) + distances0_mm[p] + distancesToEdge_mm[p];
+
+            if (egressArea != null && !GeometryUtils.containsPoint(egressArea, pointSet.getLon(p), pointSet.getLat(p))) {
+                return true; // Point is outside supplied area, continue to next iteration.
             }
-            if (distanceTableToVertices.containsKey(edge.getToVertex())) {
-                t2 = distanceTableToVertices.get(edge.getToVertex()) + distances1_mm[p] + distancesToEdge_mm[p];
+
+            edge.seek(edges[p]);
+
+            int cost = Integer.MAX_VALUE;
+
+            if (routingVariable == RoutingVariable.DISTANCE_MILLIMETERS) {
```

Comment:
I see that the distance to point map is being generalized to a cost to point map, introducing different code paths for distance and duration. I'm still trying to contrast with how we handled this dichotomy before. This is a section I'd like to discuss.

Reply:
Additional info from @ansoncfit: this is due to factoring out some repetitive code dealing with creating packed cost/distance arrays that appeared in several places. The two cases were apparently handled in separate copies of similar code before the refactoring in this PR.